pkg/destroy/ibmcloud/disk.go

```diff
@@ -0,0 +1,84 @@
+package ibmcloud
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func (o *ClusterUninstaller) listDisks() ([]cloudResource, error) {
+	o.Logger.Infof("Listing disks")
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+
+	options := o.vpcSvc.NewListVolumesOptions()
+	resources, _, err := o.vpcSvc.ListVolumesWithContext(ctx, options)
+	if err != nil {
+		return nil, errors.Wrap(err, "Listing disks failed")
+	}
+
+	result := []cloudResource{}
+	clusterOwnedTag := o.clusterLabelFilter()
+
+	for _, volume := range resources.Volumes {
+		userTags := strings.Join(volume.UserTags, ",")
+		if strings.Contains(userTags, clusterOwnedTag) {
+			o.Logger.Infof("Found disk: %s", *volume.Name)
+			result = append(result, cloudResource{
+				key:      *volume.ID,
+				name:     *volume.Name,
+				status:   *volume.Status,
+				typeName: "disk",
+				id:       *volume.ID,
+			})
+		}
+	}
+
+	return result, nil
+}
+
+func (o *ClusterUninstaller) deleteDisk(item cloudResource) error {
+	o.Logger.Infof("Deleting disk %s", item.name)
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+	options := o.vpcSvc.NewDeleteVolumeOptions(item.id)
+	details, err := o.vpcSvc.DeleteVolumeWithContext(ctx, options)
+
+	if err != nil && details.StatusCode == http.StatusNotFound {
+		// The resource is gone
+		o.deletePendingItems(item.typeName, []cloudResource{item})
+		o.Logger.Infof("Deleted disk %s", item.name)
```

Comment:
Considering the disk is deleted here, is it ok to return from here ?

Reply:
We can but we are doing return nil once at the end for success case.